hal/src/thumbv6m/sercom/v2/uart/capability.rs

```diff
@@ -0,0 +1,278 @@
+//! Capability
+//!
+//! Type-level constructs for specifiying the capabilities of a UART peripheral
+
+use crate::typelevel::Sealed;
+use bitflags::bitflags;
+use core::convert::TryFrom;
+
+/// Type-level enum representing the capabilities of a UART peripheral
+pub trait Capability: Sealed {
+    const FLAG_MASK: u8;
+    const STATUS_MASK: u16;
+}
+
+/// Type-level enum representing a UART that can transmit
+pub trait Transmit: Capability {}
+
+/// Type-level enum representing a UART that can receive
+pub trait Receive: Capability {}
+
+/// Type-level enum representing a UART that has transmit or receive
+/// capability, but not both
+pub trait Simplex: Capability {}
+
+/// Marker type representing a UART that has both transmit and receive
+/// capability
+pub enum Duplex {}
+
+impl Sealed for Duplex {}
+
+impl Capability for Duplex {
+    // All flags are valid for a Duplex UART
+    const FLAG_MASK: u8 =
+        // DRE
+        0x01 |
+        // TXC
+        0x02 |
+        // RXC
+        0x04 |
+        // RXS
+        0x08 |
+        // CTSIC
+        0x10 |
+        // RXBRK
+        0x20 |
+        // ERROR
+        0x80;
```

Comment:
Maybe you could express this as a `Status` type and then convert it to a `u8`? That might be clearer than all the comments. I'm not sure if the conversion to a `u8` is a `const` function or not though.

Reply:
Unfortunately I tried doing that, and it seems that the flags served out by `bitflags` aren't actually `const` friendly. I'll play a bit more with it, but it might have to stay a plain `u8`. It would make for a good feature suggestion for `bitflags` though.

Reply:
Hmm, it looks like the `.bits()` function is indeed `const`. But I don't see any way to create the value, because the `BitOr` trait function isn't `const`. And I don't see any other way to construct the correct value that is `const`.

Reply:
Looks like this is a workaround.

https://github.com/bitflags/bitflags/issues/180#issuecomment-499302965

But I don't know how much I like it compared to the manual approach.

Reply:
Whoops, you were commenting as I was reviewing.